pkg/cloud/talos/provisioners/azure/azure.go

```diff
@@ -0,0 +1,416 @@
+	vm := compute.VirtualMachine{
+		Location: to.StringPtr(azureConfig.Location),
+		VirtualMachineProperties: &compute.VirtualMachineProperties{
+			HardwareProfile: &compute.HardwareProfile{
+				VMSize: compute.VirtualMachineSizeTypes(azureConfig.Instances.Type),
+			},
+			StorageProfile: &compute.StorageProfile{
+				ImageReference: &compute.ImageReference{ID: to.StringPtr(azureConfig.Instances.Image)},
+				OsDisk: &compute.OSDisk{
+					OsType:       compute.Linux,
+					Name:         to.StringPtr(machine.ObjectMeta.Name + "-os-disk"),
+					CreateOption: compute.DiskCreateOptionTypesFromImage,
+					DiskSizeGB:   to.Int32Ptr(int32(azureConfig.Instances.Disks.Size)),
+				},
+			},
+			OsProfile: &compute.OSProfile{
+				ComputerName:  to.StringPtr(machine.ObjectMeta.Name),
+				AdminUsername: to.StringPtr(user),
+				AdminPassword: to.StringPtr(password),
+				CustomData:    to.StringPtr(udb64),
+			},
+			NetworkProfile: &compute.NetworkProfile{
+				NetworkInterfaces: &[]compute.NetworkInterfaceReference{
+					{
+						ID: nicObject.ID,
+						NetworkInterfaceReferenceProperties: &compute.NetworkInterfaceReferenceProperties{
+							Primary: to.BoolPtr(true),
+						},
+					},
+				},
+			},
+		},
+	}
+
+	vmClient, err := vmclient()
+	if err != nil {
+		return err
+	}
+
+	// Keeping this here for posterity. Maybe we wanna wait for completion at some point
+	// vmfuture, err := vmClient.CreateOrUpdate(ctx, azureConfig.ResourceGroup, machine.ObjectMeta.Name, vm)
+	// if err != nil {
+	// 	return err
+	// }
+	// err = vmfuture.WaitForCompletionRef(ctx, vmClient.Client)
+	// if err != nil {
+	// 	return err
+	// }
+
+	_, err = vmClient.CreateOrUpdate(ctx, azureConfig.ResourceGroup, machine.ObjectMeta.Name, vm)
+	if err != nil {
+		return err
+	}
+
+	log.Println("[Azure] Instance created: " + machine.ObjectMeta.Name)
+
+	return nil
+}
+
+//Update updates a given Azure instance.
+func (azure *Az) Update(ctx context.Context, cluster *clusterv1.Cluster, machine *clusterv1.Machine, clientset *kubernetes.Clientset) error {
+	return nil
+}
+
+// Delete deletes a Azure instance.
+func (azure *Az) Delete(ctx context.Context, cluster *clusterv1.Cluster, machine *clusterv1.Machine, clientset *kubernetes.Clientset) error {
+
+	machineSpec, err := utils.MachineProviderFromSpec(machine.Spec.ProviderSpec)
+	if err != nil {
+		return err
+	}
+
+	azureConfig := &ClusterInfo{}
+	yaml.Unmarshal([]byte(machineSpec.Platform.Config), azureConfig)
+
+	vmClient, err := vmclient()
+	if err != nil {
+		return err
+	}
+
+	// Submit the delete request
+	vmfuture, err := vmClient.Delete(ctx, azureConfig.ResourceGroup, machine.ObjectMeta.Name)
+	if err != nil {
+		return err
+	}
+
+	err = vmfuture.WaitForCompletionRef(ctx, vmClient.Client)
+	if err != nil {
+		return err
+	}
+
+	// Cleanup os disk
+	disksClient, err := disksclient()
+	if err != nil {
+		return err
+	}
+	_, err = disksClient.Delete(ctx, azureConfig.ResourceGroup, machine.ObjectMeta.Name+"-os-disk")
```

Comment:
Ignoring the deletion future?

Reply:
or, maybe you don't actually care? Good hygiene, but I could see a case for ignoring failure here.